cloudinit/distros/parsers/ifconfig.py

```diff
@@ -0,0 +1,257 @@
+# Copyright(C) 2022 FreeBSD Foundation
+#
+# Author: Mina Galić <me+FreeBSD@igalic.co>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import copy
+import re
+from functools import lru_cache
+from ipaddress import IPv4Address, IPv4Interface, IPv6Interface
+from typing import Dict, Optional, Tuple
+
+from cloudinit import log as logging
+
+LOG = logging.getLogger(__name__)
+
+
+class Ifstate:
```

Comment:
this whole file is BSD specific, and the reason is because Linux has long since deprecated ifconfig
but also, neither Linux' ifconfig nor its ip provide the depth of information that the BSD utilities do.

ifconfig on BSD is a userland tool to expose kernel information, and is treated to require the same kind of stability as an ABI. new things might be added, but things are changed or removed with great care.

my overconfident guess is that the general form of this output hasn't changed in a decade on each of the BSSs